File: src/main/intellibitz/sted/widgets/GCButton.java
package intellibitz.sted.widgets;

import javax.swing.*;
import java.awt.*;
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;

public class GCButton
        extends JButton
        implements ActionListener {
    private final Runtime runtime;

    public GCButton(Icon icon, Icon rollOverIcon) {
        super();
        setIcon(icon);
        setRolloverIcon(rollOverIcon);
        setRolloverEnabled(true);
        setMargin(new Insets(0, 0, 0, 0));
        setSize(new Dimension(icon.getIconHeight(), icon.getIconWidth()));
        setToolTipText("Runs Garbage Collector");
        runtime = Runtime.getRuntime();
        // add the button as the self action listener
        // the button knows what to do anyways
        addActionListener(this);
    }

    public void actionPerformed(ActionEvent e) {
        runtime.runFinalization();
        runtime.gc();
    }

}

File: src/main/intellibitz/sted/widgets/GCButton.kt
package sted.widgets

import java.awt.Dimension
import java.awt.Insets
import java.awt.event.ActionEvent
import java.awt.event.ActionListener
import javax.swing.Icon
import javax.swing.JButton

class GCButton : JButton(), ActionListener {
    override fun actionPerformed(e: ActionEvent) {
        val runtime = Runtime.getRuntime()
        runtime.runFinalization()
        runtime.gc()
    }

    fun load(icon: Icon?, rollOverIcon: Icon?) {
        isRolloverEnabled = true
        margin = Insets(0, 0, 0, 0)
        toolTipText = "Runs Garbage Collector"
        // add the button as the self action listener
        // the button knows what to do anyways
        addActionListener(this)
        if (icon != null) {
            setIcon(icon)
            size = Dimension(icon.iconHeight, icon.iconWidth)
        }
        rolloverIcon = rollOverIcon
    }
}